pkg/apps/graph/edges.go

```diff
@@ -92,7 +92,11 @@ func AddAllDeploymentEdges(g osgraph.MutableUniqueGraph) {
 }
 
 func AddVolumeClaimEdges(g osgraph.Graph, dcNode *appsgraph.DeploymentConfigNode) {
-	for _, volume := range dcNode.DeploymentConfig.Spec.Template.Spec.Volumes {
+	podTemplate := dcNode.DeploymentConfig.Spec.Template
```

Comment:
This seems weird, I agree with @tnozicka here. Let's wait for the reporter to return to us with the dc details.

Reply:
Further digging I found [this code](https://github.com/openshift/origin/blob/master/pkg/apps/graph/nodes/nodes.go?#L78-L81) which worries me similarly.

Reply:
We need to get to the root cause of the problem here. What's wrong with the DC and why rather than introducing this additional check, which in all honesty should not be required nor here nor in the other place I've linked.

Reply:
Hm, in testing this patch locally, I found that:
- creating a dc with `oc create -f test/integration/testdata/test-deployment-config.yaml` and
- editing it to remove template info from its spec and saving
causes my local cluster to crash with a nil pointer dereference.
iow, I do not receive the error `deploymentconfigs ... was not valid: * spec.template: Required value`, but instead I'm allowed to patch an invalid dc

http://pastebin.test.redhat.com/545803

Can go ahead open issue + PR to address this.

Reply:
@juanvallejo yes, please go ahead with the issue

Reply:
Opened issue here: https://github.com/openshift/origin/issues/18069

Reply:
Apparently the bug from https://github.com/openshift/origin/issues/18069 does not relate to the one this PR attempts to resolve. The panic was occurring during server-side validation of a deplymentconfig's deployment strategy, meaning that it happened before the invalid resource would have been successfully created or patched.

Reply:
@juanvallejo @mfojtik that's good news because that would have been pretty serious if we had those invalid DCs stored in etcd